### ISSUE TITLE
Some values in `dart_test.yaml` are ignored when running `flutter test`

### ISSUE THREAD

**Issue by WorikQCI** [flutter/flutter#113507]
tl;dr Is there an equivelent of `dart_test.yaml` for `flutter test`?

This was going to be a bug report that `reporter` and `timeout` have no effect in `dart_test.yaml`.

But I am testing a flutter app.  Using `flutter test` not `dart test`

I have been using the [dart testing docs](https://github.com/dart-lang/test/tree/master/pkgs/test/doc) and most of it applies to `flutter test`.  But it is different (such as ignoring (??) `dart_test.yaml`, at least as far as `timeout` and `reporter` go).

Is there proper documentation for `flutter test`?  https://github.com/flutter/tests is not like https://github.com/dart-lang/test, it is something different and less useful for testing flutter apps.

https://docs.flutter.dev/testing is not what I am looking for.  I know what tests I want to do, I know how I want to implement them, I would just like some documentation of the tool.  I have not found that yet.

**Comment by WorikQCI** [comment]
Ok.  This is a bug.

`flutter test` does use `dart_test.yaml`.

It ignores some things and insists on others (at least one other)

Using tags (`flutter test -t <some flag>`) if the tag is not declared in `dart_test.yaml` (despite the name) it fails:

```
 % flutter test -t backend_test -r expanded -j 1
Error on line 7, column 5 of dart_test.yaml: tags must be a map.
  ╷
7 │ tags:
  │     ^
  ╵
```

I have no expected behaviour, except documentation and consistency.    Clearly I am misusing something, and perhaps there is a bug.

**Comment by danagbemava-nc** [comment]
Hi @WorikQCI,

Can you please provide your `flutter doctor -v` and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?

Please also provide the complete output of the command that you run. 

Thank you

**Comment by WorikQCI** [comment]
> Hi @WorikQCI,
> 
> Can you please provide your `flutter doctor -v` and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?
> 
> Please also provide the complete output of the command that you run.
> 
> Thank you

Not really relevant.  

```
mport 'package:test/test.dart';

// https://github.com/microsoft/vscode/issues/163802#issue-1410731798
void main() async {
  //
  int foo = 1;
  test("Start string $foo " "  end string", () async {
    expect(true, true);
  });
}
```

That actually is a piece of code to illustrate a bug in Visual Studio Code.  But it is a test, and using `flutter test` this in `dart_test.yaml` will be ignored (it is not ignored for `dart test`

```
## These are ignored by `flutter test` but not `dart test`
reporter: expanded
timeout: 1s
```

Output of doctor:
```
[✓] Flutter (Channel stable, 3.3.4, on macOS 12.6 21G115 darwin-arm (Rosetta), locale en-AU)
    • Flutter version 3.3.4 on channel stable at /Users/user/Documents/Flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision eb6d86ee27 (13 days ago), 2022-10-04 22:31:45 -0700
    • Engine revision c08d7d5efc
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13F17a
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.72.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.50.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.6 21G115 darwin-arm (Rosetta)
    • Chrome (web)    • chrome • web-javascript • Google Chrome 106.0.5249.119
    ! Error: IPadOfWorik has recently restarted. Xcode will continue when IPadOfWorik is unlocked. (code -14)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

**Comment by danagbemava-nc** [comment]
Hi @WorikQCI, kindly also share the sample `dart_test.yaml` you used. 

Can you also update the title of the issue to reflect what you're reporting?

Thank you

**Comment by WorikQCI** [comment]
> 


@danagbemava-nc 
> Hi @WorikQCI, kindly also share the sample `dart_test.yaml` you used.

I missed this request, sorry.  Only saw the next one.

```
# This package's tests are very slow. Double the default timeout.
# timeout: 2x

# This is a browser-only package, so test on chrome by default.
# platforms: [vm]

tags:
  # # Integration tests are even slower, so increase the timeout again.
  # integration: {timeout: 2x}
  backend_test:
  heuristic_test:
  interactive_test:
  issue_demo: # Tests that illustrate an issue

## These are ignored by `flutter test` but not `dart test`
reporter: expanded
timeout: 1s
```

**Comment by danagbemava-nc** [comment]
I can reproduce the issue on `stable` and `master` using the dart_test.yaml file in  https://github.com/flutter/flutter/issues/113507#issuecomment-1299329059 and the test in https://github.com/flutter/flutter/issues/113507#issuecomment-1281399491

Possibly related: https://github.com/flutter/flutter/issues/62966

Running the test with `flutter test` just prints `All tests passed` whereas running `dart test` outputs `00:00 +0: test/widget_test.dart: Start string 1   end string` in addition to `All tests passed`. 

Labeling for further investigation.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.3.6, on macOS 13.0 22A380 darwin-arm, locale en-GB)
    • Flutter version 3.3.6 on channel stable at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6928314d50 (7 days ago), 2022-10-25 16:34:41 -0400
    • Engine revision 3ad69d7be3
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A309
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9014738/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.72.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.0 22A380 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 107.0.5304.87

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

```
[!] Flutter (Channel master, 3.5.0-12.0.pre.73, on macOS 13.0 22A380 darwin-arm64, locale en-GB)
    • Flutter version 3.5.0-12.0.pre.73 on channel master at /Users/nexus/dev/sdks/flutters
    ! Warning: `flutter` on your path resolves to /Users/nexus/dev/sdks/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutters. Consider adding /Users/nexus/dev/sdks/flutters/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/nexus/dev/sdks/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nexus/dev/sdks/flutters. Consider adding /Users/nexus/dev/sdks/flutters/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4b31473866 (69 minutes ago), 2022-11-02 01:53:15 -0400
    • Engine revision f9abfc5964
    • Dart version 2.19.0 (build 2.19.0-361.0.dev)
    • DevTools version 2.19.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A309
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9014738/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.72.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.0 22A380 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 107.0.5304.87

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 3 categories.

```
</details>

**Comment by christopherfujino** [comment]
cc @eliasyishak
